### ISSUE TITLE
1.10 --after importing workflow, got js error and diagram broken

### ISSUE THREAD

**Issue by infofromca** [OrchardCMS/Orchard#6390]
![login-wf](https://cloud.githubusercontent.com/assets/297675/13031899/85d3b1fc-d2ac-11e5-9256-9a71e0ece036.png)

**Comment by infofromca** [comment]
for whole site's recipe , it is ok.
but when i only export,import the workflows and dynamic forms, got the error

**Comment by BenedekFarkas** [comment]
The API call to RenderActivity returns a 404, which happens when an activity in the workflow is not available. This can happen if that activity is in a module that is not available (or in a feature that is not enabled). Please check the activities of that workflow in the recipe and see if you can find any activities that you couldn't add manually to a new workflow - that means that it's not available, so you can't render it either in the imported workflow.

This kind of error should be handled differently, e.g. render the missing activity it with some default visuals (and maybe a red tile) and mark it as unavailable.

**Comment by sebastienros** [comment]
Apparently @BenedekFarkas could not repro the issue. If you have some repro with a simpler case from a vanilla Orchard setup, please reopen it.

**Comment by dcinzona** [comment]
I'm seeing this as well.  Let's reopen this please.  This was not with an import, but upgrading our production environment to 1.10 from 1.9.2

![screen shot 2016-02-24 at 12 38 19 pm](https://cloud.githubusercontent.com/assets/52411/13294766/7f741ef6-daf3-11e5-8718-84bdc264c630.png)
The workflow fails to execute:
```
System.NullReferenceException: Object reference not set to an instance of an object.
   at Orchard.Workflows.Services.WorkflowManager.ExecuteWorkflow(WorkflowContext workflowContext, ActivityRecord activityRecord, IDictionary`2 tokens)
   at Orchard.Workflows.Services.WorkflowManager.StartWorkflow(WorkflowContext workflowContext, ActivityRecord activityRecord, IDictionary`2 tokens)
   at Orchard.Workflows.Services.WorkflowManager.TriggerEvent(String name, IContent target, Func`1 tokensContext)
   at Orchard.Events.DelegateHelper.<>c__DisplayClass16_0`4.<BuildAction>b__0(Object target, Object[] p)
   at Orchard.Events.DefaultOrchardEventBus.TryInvokeMethod(IEventHandler eventHandler, Type interfaceType, String messageName, String interfaceName, String methodName, IDictionary`2 arguments, IEnumerable& returnValue)
   at Orchard.Events.DefaultOrchardEventBus.TryNotifyHandler(IEventHandler eventHandler, String messageName, String interfaceName, String methodName, IDictionary`2 eventData, IEnumerable& returnValue)
```

**Comment by dcinzona** [comment]
Looks like some Workflow features were disabled after the update, that were required by the workflow:
These are the features I had to enable to get the workflows to work again:

  * Email Workflows Activities 
  * Roles Workflows Activities 
  * Users Workflows Activities 
  * Workflows Timer

**Comment by BenedekFarkas** [comment]
Timer was moved to a different sub-feature at some point after 1.8. Some other activities were also placed in separate features so that enabling Workflows doesn't need e.g. Roles if you don't use the Roles activities.

**Comment by dcinzona** [comment]
We use roles and email, both were requires but not enabled after updating to 1.10.  They were previously available without having to enable them separately.  Not sure if you consider this a breaking change, but maybe, since it broke our workflows.

**Comment by mjy78** [comment]
I agree with @dcinzona this could be a breaking change. I noticed the new Email Workflows Activities feature in 1.9.2, but even though it wasn't enabled, our email activities were still working. This was only fortunate because of a bug which has since been fixed in 1.9.3 by this commit https://github.com/OrchardCMS/Orchard/commit/15ccba2b8122e002d6f1fac8fd83d1aad6de7ecc

I've already gone through 30+ sites in our multi-tenant installation and manually enabled the "Email Worflows Activity" feature in anticipation of the upgrade to 1.9.3 to ensure our workflows don't break. At a minimum there should be a note about needing to do this in the release notes. Ideally the upgrade should enable this feature automatically for any tenants that already have the email module enabled.
